code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -45,17 +45,137 @@
 	return 1	//No drifting in space for space carp!	//original comments do not steal
 
 /mob/living/simple_animal/hostile/carp/FindTarget()
-	. = ..()
-	if(.)
-		custom_emote(1,"nashes at [.]")
+	if(!faction) //No faction, no reason to attack anybody.
+		return null
+
+	var/atom/T = null
+	var/lowest_health = INFINITY // Max you can get
+	stop_automated_movement = 0
+	for(var/mob/living/L in targets)
+		if((L.faction == src.faction) && !attack_same)
+			continue
+		if(L in friends)
+			continue
+
+		if(!L.stat && (L.health < lowest_health))
+			lowest_health = L.health
+			T = L
+	if(isnull(T))
+		for(var/atom/A in targets)
+
+			if(A == src)
+				continue
+
+			var/atom/F = Found(A)
+			if(F)
+				T = F
+				break
+
+			else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
+				var/obj/mecha/M = A
+				if (M.occupant)
+					T = M
+					break
+
+			if(istype(A, /obj/machinery/bot))
+				var/obj/machinery/bot/B = A
+				if (B.health > 0)
+					T = B
+					break
+
+			if(istype(A, /obj/effect/energy_field))
+				T = A
+				break
+
+	if (T != target_mob)
+		target_mob = T
+		FoundTarget()
+		custom_emote(1,"nashes at [target_mob]")
+	if(!isnull(T))
+		stance = HOSTILE_STANCE_ATTACK
+	return T
+
+/mob/living/simple_animal/hostile/carp/MoveToTarget()
+	stop_automated_movement = 1
+	if(istype(target_mob, /obj/effect/energy_field) && !QDELETED(target_mob) && (target_mob in targets))
+		stance = HOSTILE_STANCE_ATTACKING
+		walk_to(src, target_mob, 1, move_to_delay)
+		return 1
+	if(QDELETED(target_mob) || SA_attackable(target_mob))
+		LoseTarget()
+	if(target_mob in targets)
+		if(ranged)
+			if(get_dist(src, target_mob) <= 6)
+				OpenFire(target_mob)
+			else
+				walk_to(src, target_mob, 1, move_to_delay)
+		else
+			stance = HOSTILE_STANCE_ATTACKING
+			walk_to(src, target_mob, 1, move_to_delay)
+
+/mob/living/simple_animal/hostile/carp/AttackTarget()
+
+	stop_automated_movement = 1
+	if(istype(target_mob, /obj/effect/energy_field) && !QDELETED(target_mob) && (get_dist(src, target_mob) <= 1))
+		AttackingTarget()
+		attacked_times += 1
+		return 1
+	if(QDELETED(target_mob) || SA_attackable(target_mob))
+		LoseTarget()
+		return 0
+	if(!(target_mob in targets))
+		LoseTarget()
+		return 0
+	if(next_move >= world.time)
+		return 0
+	if(get_dist(src, target_mob) <= 1)	//Attacking
+		AttackingTarget()
+		attacked_times += 1
+		return 1
 
 /mob/living/simple_animal/hostile/carp/AttackingTarget()
-	. =..()
-	var/mob/living/L = .
-	if(istype(L))
+	setClickCooldown(attack_delay)

```

Comment:
Why is this necessary?

Reply:
That is what parent has, since it is the override